01_classify_stratified/nbconverted/plot_mutation_results.py

```diff
@@ -0,0 +1,217 @@
+axarr[1].set_ylim((0, y_max+5))
+axarr[1].legend(title=r'Reject $H_0$', loc='upper left')
+axarr[1].set_title(r'Mutation prediction, methylation data')
+
+def label_points(x, y, gene, sig, ax):
+    text_labels = []
+    a = pd.DataFrame({'x': x, 'y': y, 'gene': gene, 'sig': sig})
+    for i, point in a.iterrows():
+        if point['y'] > -np.log10(SIG_ALPHA):
+            text_labels.append(
+                ax.text(point['x'], point['y'], str(point['gene']))
+            )
+    return text_labels
+
+# label genes and adjust text to not overlap
+# automatic alignment isn't perfect, can align by hand in inkscape if necessary
+text_labels_expression = label_points(expression_results_df['delta_mean'],
+                                      expression_results_df['nlog10_p'],
+                                      expression_results_df.gene,
+                                      expression_results_df.reject_null,
+                                      axarr[0])
+adjust_text(text_labels_expression,
+            ax=axarr[0], 
+            expand_text=(1., 1.),
+            lim=5)
+
+text_labels_methylation = label_points(methylation_results_df['delta_mean'],
+                                       methylation_results_df['nlog10_p'],
+                                       methylation_results_df.gene,
+                                       methylation_results_df.reject_null,
+                                       axarr[1])
+adjust_text(text_labels_methylation,
+            ax=axarr[1],
+            expand_text=(1., 1.),
+            lim=5)
+
+
+# In[7]:
+
+
+compare_results_df = au.compare_results(methylation_df,
+                                        pancancer_df=expression_df,
+                                        identifier='identifier',
+                                        metric='aupr',
+                                        correction=True,
+                                        correction_method='fdr_bh',
+                                        correction_alpha=SIG_ALPHA,
+                                        verbose=True)
+compare_results_df.rename(columns={'identifier': 'gene'}, inplace=True)
+compare_results_df.head()
+
+
+# In[8]:
+
+
+compare_results_df['nlog10_p'] = -np.log10(compare_results_df.corr_pval)
+
+sns.set({'figure.figsize': (16, 8)})
+sns.scatterplot(data=compare_results_df, x='delta_mean', y='nlog10_p', hue='reject_null')
+plt.axvline(x=0, linestyle=':', color='black')
+l = plt.axhline(y=-np.log10(SIG_ALPHA), linestyle=':')
+plt.text(0.7, -np.log10(SIG_ALPHA)+0.05,
+         r'$\alpha = {}$'.format(SIG_ALPHA),
+         va='center', ha='center', color=l.get_color(),
+         backgroundcolor=plt.gca().get_facecolor())
+plt.xlabel('AUPR(expression) - AUPR(methylation)')
+plt.ylabel(r'$-\log_{10}($adjusted $p$-value$)$')
+plt.xlim((-0.75, 0.75))
+plt.ylim((0, y_max+5))
+plt.legend(title=r'Reject $H_0$', loc='upper left')
+plt.title(r'Cancer type prediction, expression vs. methylation')
```

Comment:
It's interesting that methylation seems to have a strictly weaker signal. Do you plan on incorporating both methylation and expression into a single model to see if it outperforms the individual models?

Reply:
Yeah, I'd definitely like to do this. Not sure if I'll get to this soon or later, but it's on the list of future directions for sure.